core/client/src/main/java/alluxio/client/file/FileOutStream.java

```diff
@@ -116,7 +120,8 @@ public FileOutStream(AlluxioURI path, OutStreamOptions options) throws IOExcepti
         String tmpPath = PathUtils.temporaryFileName(mNonce, mUfsPath);
         UnderFileSystem ufs = UnderFileSystem.get(tmpPath, ClientContext.getConf());
         // TODO(jiri): Implement collection of temporary files left behind by dead clients.
-        mUnderStorageOutputStream = ufs.create(tmpPath, (int) mBlockSize);
+        CreateOptions ufsCreateOptions = new CreateOptions().setPermission(options.getPermission());
+        mUnderStorageOutputStream = ufs.create(tmpPath, ufsCreateOptions);
```

Comment:
Is it confusing that the varaible `ufsCreateOptions` of type `CreateOptions` and not `CreateUfsFileOptions`? Maybe just name the variable `createOptions`?